integration_tests/gruntz_demo3.py

```diff
@@ -0,0 +1,184 @@
+from lpython import S, str
+from sympy import Symbol, Pow, sin, oo, pi, E, Mul, Add
+
+def mrv(e: S, x: S) -> list[S]:
+    """
+    Calculate the MRV set of the expression.
+
+    Examples
+    ========
+
+    >>> mrv(log(x - log(x))/log(x), x)
+    {x}
+
+    """
+
+    if e == x:
+        list1: list[S] = [x]
+        return list1
+    if e.func == Mul or e.func == Add:
+        a: S = e.args[0]
+        b: S = e.args[1]
+        ans1: list[S] = mrv(a, x)
+        ans2: list[S] = mrv(b, x)
+        return mrv_max(mrv(a, x), mrv(b, x), x)
+    if e.func == Pow:
+        base: S = e.args[0]
+        list3: list[S] = mrv(base, x)
+        return list3
+    if e.is_Function:
+        return reduce(lambda a, b: mrv_max(a, b, x), (mrv(a, x) for a in e.args))
```

Comment:
Blocker 1: Need to look into this as this is somewhat restrictive.

`e.is_Function` means all cases but (`log`, `exp`, `pow`, `add`, `mul`) which is basically `trig`, `inverse trig`, `hyperbolic` etc
So maybe let's just look into limits involving `sin` for now

Also have to address this line somehow `return reduce(lambda a, b: mrv_max(a, b, x), (mrv(a, x) for a in e.args))`